src/linear_algebra/affine_space/affine_subspace.lean

```diff
@@ -189,6 +189,51 @@ iff.rfl
 
 variables {k P}
 
+/-- Two affine subspaces are equal if they have the same points. -/
+@[ext] lemma coe_injective  : function.injective (coe : affine_subspace k P → set P) :=
+λ s1 s2 h, begin
+  cases s1,
+  cases s2,
+  congr,
+  exact h
+end
+
+@[simp] lemma ext_iff (s₁ s₂ : affine_subspace k P) :
+  (s₁ : set P) = s₂ ↔ s₁ = s₂ :=
+⟨λ h, coe_injective h, by tidy⟩
+
+instance : has_vadd V (affine_subspace k P) :=
+{ vadd := λ x S,
+{ carrier := x +ᵥ S,
+  smul_vsub_vadd_mem := λ v _ _ _ ha hb hc,
+  begin
+    rcases ha with ⟨a, haS, rfl⟩,
+    rcases hb with ⟨b, hbS, rfl⟩,
+    rcases hc with ⟨c, hcS, rfl⟩,
+    simp only [affine_subspace.mem_coe, vadd_vsub_vadd_cancel_left] at *,
+    refine ⟨v • (a -ᵥ b) +ᵥ c, S.smul_vsub_vadd_mem v haS hbS hcS, _⟩,
+    rw [vadd_vadd, vadd_vadd, add_comm],
+  end } }
+
+instance : add_action V (affine_subspace k P) :=
+{ vadd := (+ᵥ),
+  zero_vadd := λ p, by { unfold has_vadd.vadd, simp only [zero_vadd, set.image_id'], ext, refl, },
+  add_vadd := λ x y p,
+  begin
+    unfold has_vadd.vadd,
+    simp only [set.image_vadd],
+    ext z,
+    rw [set.mem_vadd_set, set.mem_vadd_set],
+    refine ⟨λ h, _, λ h, _⟩,
+    { rcases h with ⟨a, ha, rfl⟩,
+      refine ⟨y +ᵥ a, ⟨a, ha, rfl⟩, vadd_vadd x y a⟩, },
+    rcases h with ⟨a, ha, rfl⟩,
+    refine ⟨-y +ᵥ a, _, _⟩,
+    { rcases ha with ⟨b, hb, rfl⟩,
+      rwa [vadd_vadd, add_left_neg, zero_vadd], },
+    rw [vadd_vadd, add_neg_cancel_right],
+  end }
```

Comment:
I think these probably belong after the `coe_vadd` below; certainly, they're too high up right now. That also means you can move the `coe_injective` back.